### DISCUSSION TITLE
REST API

### DISCUSSION THREAD

**Discussion by TonyNRT** [pubpub-zz/ppInk#294] [Q&A] [answered]
@pubpub-zz I appreciate your work on ppInk! It is a great tool.
 
I am hopeful you can help me with commanding ppink actions through the REST API.

I desire to fully operate ppink while the onscreen menu is hidden using the REST API to control the pen and tool selections. I have been able to successfully command a pen and free hand tool along with delete and a few other functions. However, I am having a difficult time requesting some of the features for example "toggle fading". I am using the following requests:
"
http://localhost:7999/CurrentPen?P=0
https://localhost:7999/ToggleFading?P=-1
http://localhost:7999/CurrentTool?T=0&F=0
"
Can you explain what I am doing wrong or provide a quick example on how to fix this issue?

Could you also share with me a way to monitor the http requests and responses to ppink to better understand the state of the application and confirm it is operating as I would expect based on the requests?

Besides the the httpRequests.rtf are there any other documents providing examples or additional definition of the implementation of the standard http requests?

Thanks for you help!

**Comment by pubpub-zz** [comment, marked as answer]
@TonyNRT 
Happy you are enjoying. Do not hesitate to star the project 😀

In your request there is a mistake : the toggle is a https:// and not http://
for debugging, normally any failures will be reported : you can use any browser to test your request.
One point you may use is to disable engaging/disengaging pointer mode when using alt tab:
![image](https://github.com/pubpub-zz/ppInk/assets/4083478/2ea33c2b-c21c-43e1-a783-128c99c19347)
with this parameter off, open the toolbar, set ppink in window mode with a large enough window to see the buttons but not overlaying the address bar on your browser.
then you can test your request directly there

**Reply by TonyNRT** [reply]
@pubpub-zz 
1. Thanks for the quick identification of my error! It works now. I was referencing an older httpRequests document that listed some of the examples with "https". Newer versions of this document have the requests listed properly.

One of the http requests I am still having an issue with is the screenshot function. I have tried the following with no success:
http://localhost:7999/SnapShotScreen?A=Out
http://localhost:7999/SnapShotScreen?A=Out\
http://localhost:7999/SnapShotScreen?A=End
http://localhost:7999/SnapShotScreen?A=End\
http://localhost:7999/SnapShotScreen?A=Cont

None of these appear to activate a screenshot and there are no files added to the screenshot folder that has been identified on the ppink options menu. Is there an error in my http request or is there perhaps an issue with screenshot on 1.8.3 release? 

2. Is there a way to save the strokes as an image or pdf file graphically representing the strokes instead of the txt file?

3. The icon sizes (cursor, eraser, move, etc) are very large on the screen size I am working with. Is there a way to reduce the icon size? I see in the config.ini I am able to show/hide the icons but ultimately I would like to scale the icons down ~50%.

Thanks!

**Reply by pubpub-zz** [reply]
Oups : There is a mistake in documentation ; it should be  : http://localhost:7999/Snapshot

arguments should work
